canonical/vanilla-framework
# Indent code snippets using `js-beautify`

## Done

Formats all HTML, JS, and CSS code shown in `pre` code snippet blocks using [js-beautify](https://www.npmjs.com/package/js-beautify)

Fixes #5245, [WD-13629](https://warthogs.atlassian.net/browse/WD-13629)

## QA

- Look at some example code blocks and make sure they are indented with 2-space indentation. 
- [Full width dropdown navigation](https://vanilla-framework-5259.demos.haus/docs/patterns/navigation#full-width-dropdowns) is a good example to check as it has HTML, CSS, and JS, and was made with macros that generate some incorrect indentation in [main](https://staging.vanillaframework.io/docs/patterns/navigation#full-width-dropdowns)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

### Before
![image](https://github.com/user-attachments/assets/a6c0c970-7791-477b-8a64-d3b1cb57cca7)
### After
![image](https://github.com/user-attachments/assets/27e31576-52aa-414c-8ad8-ca80c465cffa)


[WD-13629]: https://warthogs.atlassian.net/browse/WD-13629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
